src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1284,9 +1287,9 @@ public static bool QueueUserWorkItem(WaitCallback callBack, object state, bool p
 
             ExecutionContext context = ExecutionContext.Capture();
 
-            IThreadPoolWorkItem tpcallBack = context == ExecutionContext.Default ?
-                new QueueUserWorkItemCallbackDefaultContext(callBack, state) :
-                (IThreadPoolWorkItem)new QueueUserWorkItemCallback(callBack, state, context);
+            IThreadPoolWorkItem tpcallBack = (context == null || !context.IsDefault) ?
+                new QueueUserWorkItemCallback(callBack, state, context) :
+                (IThreadPoolWorkItem)new QueueUserWorkItemCallbackDefaultContext(callBack, state);
```

Comment:
It's fine, but I'm curious why you swapped the polarity here.  Did that have an impact on something?

Reply:
So could combine the `context == null || !context.IsDefault` tests; otherwise it was a bit ugly

Reply:
I may have over thought it 😄

Reply:
Reverted